From 872319e8d09a1f19c24989636d0678371ef5143c Mon Sep 17 00:00:00 2001
From: Jiayang Mao <quic_jiaymao@quicinc.com>
Date: Tue, 26 Nov 2024 18:05:18 +0800
Subject: [PATCH v1] BACKPORT: shared/att: Fix failing to set security level

bt_att_chan_set_security attempts to set BT_SECURITY without first
checking what is the current security level which may cause errors
since the kernel does actually return -EINVAL when the security doesn't
change.

Upstream-Status: Backport [https://git.kernel.org/pub/scm/bluetooth/bluez.git/commit/?id=5c65356cae64ddbc95210deb36ee66430645edf8]

Signed-off-by: Jiayang Mao <quic_jiaymao@quicinc.com>
---
 src/shared/att.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/shared/att.c b/src/shared/att.c
index f7bef08..8340f09 100644
--- a/src/shared/att.c
+++ b/src/shared/att.c
@@ -726,6 +726,9 @@ static bool bt_att_chan_set_security(struct bt_att_chan *chan, int level)
 {
 	struct bt_security sec;
 
+	if (level == bt_att_chan_get_security(chan))
+		return true;
+
 	if (chan->type == BT_ATT_LOCAL) {
 		chan->sec_level = level;
 		return true;
-- 
2.25.1

